app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -11,12 +11,23 @@ def connect(options = {})
 
   def verify_credentials(_auth_type = nil, options = {})
     require 'azure-armrest'
-    connect(options)
-  rescue Azure::Armrest::UnauthorizedException
-    raise MiqException::MiqHostError, _("Incorrect credentials - check your Azure Client ID and Client Key")
+    conf = connect(options)
+
+    unless subscription.blank?
+      vms = ::Azure::Armrest::VirtualMachineService.new(conf)
+      subscriptions = vms.subscriptions.map(&:subscription_id)
+      unless subscriptions.include?(subscription)
+        raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - check your Azure Subscription ID")
+      end
+    end
+  rescue Azure::Armrest::UnauthorizedException, Azure::Armrest::BadRequestException
```

Comment:
Upon further review, I think this is a bad idea, because the error message is different depending on whether or not it's a bad client_id/client_key versus a bad tenant id, which just shows "invalid request".

The odds of a BadRequest happening at the moment of validation that *isn't* an actual validation error is so low, I don't think it's worth worrying about. I think we should avoid adding extra logic that's just going to get ripped out once azure-armrest 0.3.x gem is released anyway.